server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+					Type:          schema.Type,
+					CreatedByUser: schemaVersion.CreatedByUser,
+					CreationDate:  schemaVersion.CreationDate,
+				}
+				extednedSchemas = append(extednedSchemas, extSchema)
+			}
+		}
+
+	}
+
+	if len(extednedSchemas) == 0 {
+		return []models.ExtendedSchema{}, nil
+	} else {
+		return extednedSchemas, nil
+	}
+}
+
+func (sh SchemasHandler) GetAllSchemaVersionsByVersionNumber(versionNumber int) ([]models.SchemaVersion, error) {
+	var schemasVersion []models.SchemaVersion
+
+	filter := bson.M{"version_number": versionNumber}
+	cursor, err := schemasVersionCollection.Find(context.TODO(), filter)
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemasVersion); err != nil {
+		return schemasVersion, err
+	}
+
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+
+	if len(schemasVersion) == 0 {
+		return []models.SchemaVersion{}, nil
+	} else {
+		return schemasVersion, nil
+	}
+}
+
+func (sh SchemasHandler) FindAndDeleteSchema(schemaName string) error {
+	var schema models.Schema
+	filter := bson.M{"name": schemaName}
+	err := schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil {
+		return err
+	}
+	schemaVersions := schema.Versions
+	var schemaVersion models.SchemaVersion
+	for _, version := range schemaVersions {
+		filterVersion := bson.M{"_id": version}
+		err = schemasVersionCollection.FindOneAndDelete(context.TODO(), filterVersion).Decode(&schemaVersion)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = schemasCollection.DeleteOne(context.TODO(), filter)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
+	var body models.CreateNewSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.Name)
+	err := validateSchemaName(schemaName)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	exist, _, err := IsSchemaExist(schemaName)
+	if exist {
+		serv.Warnf("Schema with that name already exists")
+	}
+	if err != nil {
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("CreateNewSchema error: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
+	}
+	schemaType := strings.ToLower(body.Type)
+	err = validateSchemaType(schemaType)
+
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	schemaContent := strings.ToLower(body.SchemaContent)
+	err = validateSchemaContent(schemaContent)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	filter := bson.M{"name": schemaName}
+	var schema models.Schema
+	err = schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
```

Comment:
This check is not needed because this endpoint called only when creating new schema from scratch and not a new version, for new version we will have another endpoint, this is the edit ticket.

And in general note that mongo has special error for no documents you can take a look from the isExist function